source/kraken/realtime.cpp

```diff
@@ -46,6 +46,22 @@ namespace navitia {
 
 namespace nd = type::disruption;
 
+static bool base_vj_exists_the_same_day(const type::Data& data, const transit_realtime::TripUpdate& trip_update) {
+    const auto& mvj = *data.pt_data->get_or_create_meta_vehicle_journey(trip_update.trip().trip_id(),
```

Comment:
I understand your remark, but I already check if the meta vj exists (). So, I can call directly `get_or_create_meta_vehicle_journey`.
I save the `meta_vjs.exits(...)` call, because it scans the all meta_vj list.

Reply:
This `exists(uri)` is a `map.find()` (and even if it was a vector loop), to me the cost is not an argument good enough to keep a method that has side effect on a data argument passed as `const` (even if it's currently protected now as I previously noted, yes).
So let's use a third opinion on that to move on.